### PR TITLE
Skip region check for CCS ProjectClaims

### PR DESCRIPTION
**Pull request by NautiluX**
### What type of PR is this? 
bugfix && feature && refactor

### What this PR does / why we need it:

* In CCS, the user is responsible for ensuring enough quota in the provided project, hence skip region check
* The region check resulted in a error loop for unsupported regions, as the reconcile returned an error
* Add `ConditionInvalid` for failed condition checks
* Add missing `return` statement
* Fix some warnings for missing checks in return codes
* Refactor util.OperationResult functions for easier asserting on results
* `go mod tidy`, `go mod vendor`

### Which issue(s) this PR fixes(can be a reference to existing issue or jira/bz):

_Fixes OSD-5710_

### Pre-checks:
- [x] manually tested latest changes against crc/k8s
- [x] run the `make coverage` command to generate new calculated coverage

### CODE REVIEW

File: pkg/controller/projectclaim/projectclaimadapter.go [hunk 1]
```diff
@@ -203,7 +204,7 @@ func (c *ProjectClaimAdapter) EnsureProjectClaimInitialized() (gcputil.Operation
 		c.projectClaim.Status.Conditions = []gcpv1alpha1.Condition{}
 		err := c.client.Status().Update(context.TODO(), c.projectClaim)
 		if err != nil {
-			gcputil.RequeueWithError(operrors.Wrap(err, "failed to initalize projectclaim"))
+			return gcputil.RequeueWithError(operrors.Wrap(err, "failed to initalize projectclaim"))
```
Comment (rafael-azevedo) [inline pkg/controller/projectclaim/projectclaimadapter.go:207]: nice catch

File: pkg/controller/projectclaim/projectclaimadapter.go [hunk 2]
```diff
@@ -312,35 +313,36 @@ func (c *ProjectClaimAdapter) EnsureProjectClaimState(state gcpv1alpha1.ClaimSta
...
 }
 
 // SetProjectClaimCondition calls SetCondition() with project claim conditions
-func (c *ProjectClaimAdapter) SetProjectClaimCondition(reason string, err error) error {
+func (c *ProjectClaimAdapter) SetProjectClaimCondition(conditionType gcpv1alpha1.ConditionType, reason string, err error) (gcputil.OperationResult, error) {
 	conditions := &c.projectClaim.Status.Conditions
-	conditionType := gcpv1alpha1.ConditionError
 	if err != nil {
 		c.conditionManager.SetCondition(conditions, conditionType, corev1.ConditionTrue, reason, err.Error())
-	} else {
```
Comment (rafael-azevedo) [inline pkg/controller/projectclaim/projectclaimadapter.go:329]: this makes me so happy

File: pkg/controller/projectclaim/projectclaimadapter.go [hunk 3]
```diff
@@ -350,12 +352,14 @@ func (c *ProjectClaimAdapter) IsRegionSupported() error {
...
 func (c *ProjectClaimAdapter) EnsureRegionSupported() (gcputil.OperationResult, error) {
-	if err := c.IsRegionSupported(); err != nil {
+	err := c.IsRegionSupported()
+	if err != nil {
 		c.projectClaim.Status.State = gcpv1alpha1.ClaimStatusError
-		c.StatusUpdate()
-		return gcputil.RequeueWithError(operrors.Wrap(err, ""))
 	}
-	return gcputil.ContinueProcessing()
+	if err == nil && c.projectClaim.Status.State == gcpv1alpha1.ClaimStatusError {
```
Comment (rafael-azevedo) [inline pkg/controller/projectclaim/projectclaimadapter.go:359]: why are we setting the state to Pending if there is a ClaimStatausError ?

File: pkg/controller/projectclaim/projectclaimadapter.go [hunk 4]
```diff
@@ -350,12 +352,14 @@ func (c *ProjectClaimAdapter) IsRegionSupported() error {
...
+		c.projectClaim.Status.State = "Pending"
```
Comment (rafael-azevedo) [inline pkg/controller/projectclaim/projectclaimadapter.go:360]: Is there no projectclaim status type for pending ?

File: pkg/controller/projectclaim/projectclaimadapter.go (hunk 3 above)
Comment (NautiluX) [reply pkg/controller/projectclaim/projectclaimadapter.go:359]: if it was error before, we set it back to pending if it is resolved

File: pkg/controller/projectclaim/projectclaimadapter.go (hunk 4 above)
Comment (NautiluX) [reply pkg/controller/projectclaim/projectclaimadapter.go:360]: there is. Good catch!

File: pkg/controller/projectclaim/projectclaimadapter.go (hunk 2 above)
Comment (NautiluX) [reply pkg/controller/projectclaim/projectclaimadapter.go:329]: I know :)

**Review by rafael-azevedo — APPROVED** [review]
/lgtm

**Pull request merged**